### PR TITLE
MCR-2856 JobQueue improvements

### PR DESCRIPTION
**Pull request by sebhofmann**
* avoid naming things like slaveFactory
* reduced complexity of MCRJobManager#run method and other methods
* moved most configuration options to interface
* added MAX_TRIES status, exception and tries collumns to database
* change job status to MAX_TRIES after they run n times (configurable)
* added commands to list and reset jobs which have job status MAX_TRIES
* removed single queue mode (makes code more complex and nobody should have used it)
* added more tests
* do not reset processing jobs every x minutes, reset them once at startup
* extracted cronjob code from base to new module required adding `<mapping-file>META-INF/mycore-jobqueue-mappings.xml</mapping-file>` to persistence.xml
* moved jersey code to rest api endpoint /api/jobqueue
* added vue based gui for jobqueues

[Link to jira](https://mycore.atlassian.net/browse/MCR-2856).

### CODE REVIEW

**Review by Mewel — CHANGES_REQUESTED** [review]
* MCRJobRunnable#run() is to long
* MCRJobRunnable#run() setStatus(MCRProcessableStatus.failed) is never set
* MCRJobThreadStarter constructor is to long -> should use static inner classes for MCRJobQueueEventListener and ThreadPoolExecutor
* MCRTestJobAction class should be moved to test -> maybe rename because there is already one in test
* MCRJobThreadStarterTest#testRun() 87 -> try to avoid waiting in test cases if possible -> this is very annoying for building mycore

minor code improvements:
* some LOGGER's and other member variables maybe final
* MCRJobQueue 226 -> return Collections.emptyIterator();
* MCRJobResetter 45 -> private final Function<Class<?>, Queue<MCRJob>> queueResolver;
* MCRJobThreadStarter 203 -> null check can be removed
* MCRJobDAOJPAImplTest 271 and multiple others -> allJobs.forEach(em::persist);
MCRJobQueueTest 48 -> MCRJobQueueEventListener listener = notifiedJobs::add;

**Comment by sebhofmann** [comment]
> * MCRJobRunnable#run() is to long

I extracted handleJobExecutionException

> * MCRJobRunnable#run() setStatus(MCRProcessableStatus.failed) is never set

Its set by setError(Throwable method)

> * MCRJobThreadStarter constructor is to long -> should use static inner classes for MCRJobQueueEventListener and ThreadPoolExecutor

Extracted ListenerNotifier, JobThreadFactory and ActiveCountingThreadPoolExecutor to static inner class 

> * MCRTestJobAction class should be moved to test -> maybe rename because there is already one in test
> * MCRJobThreadStarterTest#testRun() 87 -> try to avoid waiting in test cases if possible -> this is very annoying for building mycore

Removed "Production" TestJobAction and related Command

> 
> minor code improvements:
> 
> * some LOGGER's and other member variables maybe final

Set all loggers to final

> * MCRJobQueue 226 -> return Collections.emptyIterator();

Done

> * MCRJobResetter 45 -> private final Function<Class<?>, Queue> queueResolver;

Changed to ? extends MCRJobAction

> * MCRJobThreadStarter 203 -> null check can be removed

Done

> * MCRJobDAOJPAImplTest 271 and multiple others -> allJobs.forEach(em::persist);

Done 

>   MCRJobQueueTest 48 -> MCRJobQueueEventListener listener = notifiedJobs::add;

Done

File: mycore-jobqueue/src/main/java/org/mycore/services/queuedjob/MCRJobRunnable.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    }
+
+    private void handleSuccess(MCRJob localJob) {
+        localJob.setStatus(MCRJobStatus.FINISHED);
+        setStatus(MCRProcessableStatus.successful);
+    }
+
+    private void handleException(MCRJob localJob, Exception executionException) {
+        try (StringWriter sw = new StringWriter(); PrintWriter pw = new PrintWriter(sw);) {
+            executionException.printStackTrace(pw);
```
[the remark this answers is not in the document]
Comment (yagee-de) [reply mycore-jobqueue/src/main/java/org/mycore/services/queuedjob/MCRJobRunnable.java:169]: @sonatype-lift ignore

**Review by yagee-de — APPROVED** [review]
change requests from @Mewel applied and added changes for sonatype and organized imports

**Pull request merged**